cobbler/cobbler_collections/collection.py

```diff
@@ -432,15 +430,13 @@ def add(self, ref, save: bool = False, with_copy: bool = False, with_triggers: b
                 utils.run_triggers(self.api, ref, "/var/lib/cobbler/triggers/add/%s/post/*" % self.collection_type(),
                                    [])
 
-    def __duplication_checks(self, ref, check_for_duplicate_names: bool, check_for_duplicate_netinfo: bool):
+    def __duplication_checks(self, ref, check_for_duplicate_names: bool):
```

Comment:
If we only check for duplicate names, then this can be made much much easier. Thus I believe there is work do be done. Because a name is something all items have, the collection type doesn't matter. Thus just loop through the list of items and check for the name. We DON'T want to use the key of the dict (which would be insanely fast) because we want to change that to the UID as of version 3.4.0 (maybe even 3.3.1).

Reply:
Not sure I understand. I also believe the duplicate name checking can get enhanced and all this can be made somewhat cleaner. Possibly also with your getter/setter approach, not sure about the names right now.

Also the save= param which is passed from function to function in an ugly way, probably could be done nicer at a later point.

But this patch is about removing check_for_duplicate_netinfo which is already done via your new ip, ip_v6, dns_name and mac setters...

So this is the way to go, right?